sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -2695,4 +2695,26 @@ class SubquerySuite extends QueryTest
       }
     }
   }
+
+  test("SPARK-42937: Outer join with subquery in condition") {
+    withSQLConf(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "false",
```

Comment:
Why does this fail only when AQE is disabled?

Reply:
@allisonwang-db 

When adaptive execution is enabled, `PlanAdaptiveSubqueries` always sets `shouldBroadcast=true`, so the subquery's result is available on the executor, if needed.

https://github.com/apache/spark/blob/31965a06c9f85abf2296971237b1f88065eb67c2/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/PlanAdaptiveSubqueries.scala#L46